src/Fbgemm.cc

```diff
@@ -136,28 +137,42 @@ void fbgemmPacked(
   t_very_start = std::chrono::high_resolution_clock::now();
 #endif
 
+  thread_type_t thread_partition_info =
+      fbgemmGetThreadPartition(MDim, NDim, thread_id, num_threads);
```

Comment:
We first do 2D partition assuming G == 1 and then partition across m when there're multiple groups after the fact. We may want to first decide how many threads will be used for each group and then 2D-partition based on that? In this way, we should be able to do globally better partitioning.

Reply:
Created a "3D" thread partition function to select all the thread partition number across g, m, and n dimensions.